src/math.ts

```diff
@@ -147,13 +147,29 @@ export const centerPoint = (a: Point, b: Point): Point => {
 // to be considered a loop
 export const isPathALoop = (
   points: ExcalidrawLinearElement["points"],
+  /** supply if you want the loop detection to account for current zoom */
+  zoomValue: Zoom["value"] = 1 as NormalizedZoomValue,
 ): boolean => {
   if (points.length >= 3) {
     const [firstPoint, lastPoint] = [points[0], points[points.length - 1]];
-    return (
-      distance2d(firstPoint[0], firstPoint[1], lastPoint[0], lastPoint[1]) <=
-      LINE_CONFIRM_THRESHOLD
+    const distance = distance2d(
+      firstPoint[0],
+      firstPoint[1],
+      lastPoint[0],
+      lastPoint[1],
     );
+
+    // Adjusting LINE_CONFIRM_THRESHOLD to current zoom so that when zoomed in
+    // really close we make the threshold smaller, and vice versa.
+    // The formulas are a result of curve-fitting target values.
```

Comment:
```suggestion
    // Adjusting LINE_CONFIRM_THRESHOLD to current zoom so that when zoomed in
    // really close we make the threshold smaller, and vice versa.
    // The formula is a result of mostly-manual curve-fitting values which I felt
    // appropriate for given zoom values (0.1..55px, 1..8px, 10..2px)
```